chromadb/utils/embedding_functions.py

```diff
@@ -106,3 +106,29 @@ def __init__(self, model_name: str = "hkunlp/instructor-base", device="cpu"):
 
     def __call__(self, texts: Documents) -> Embeddings:
         return self._model.encode(texts).tolist()
+
+
+class GooglePalmEmbeddingFunction(EmbeddingFunction):
+    """To use this EmbeddingFunction, you must have the google.generativeai Python package installed and have a PaLM API key.
+    """
+    def __init__(self, api_key: str, model_name: str):
+        if not api_key:
+            raise ValueError("Please provide a PaLM API key.")
+
+        if not model_name:
+            raise ValueError("Please provide the model name.")
+
+        try:
+            import google.generativeai as palm
+        except ImportError:
+            raise ValueError("The Google Generative AI python package is not installed.")
```

Comment:
Make consistent with other error messages - instruct the user about how to install.